internal/handlers/jsonb1/projection.go

```diff
@@ -30,23 +36,182 @@ func projection(projection *types.Document, p *pg.Placeholder) (sql string, args
 		return
 	}
 
-	ks := ""
-	for i, k := range projection.Keys() {
-		if i != 0 {
-			ks += ", "
-		}
-		ks += p.Next()
-		args = append(args, k)
+	// create a list of keys for document
+	ks, arg, err := s.buildProjectionKeys(projection.Keys(), projectionMap, p)
+	if err != nil {
+		err = lazyerrors.Errorf("buildProjectionKeys: %w", err)
+		return
 	}
-	sql = "json_build_object('$k', array[" + ks + "],"
-	for i, k := range projection.Keys() {
-		if i != 0 {
+	args = append(args, arg...)
+	sql = "json_build_object('$k', array[" + ks + "], "
+
+	// _id and _id value
+	sql += p.Next() + "::text, _jsonb->" + p.Next() + " "
+	args = append(args, "_id", "_id")
+
+	// build json object
+	for _, k := range projection.Keys() { // value
+		s.l.Sugar().Debugf("projection key %s", k)
+
+		doc, isDoc := projectionMap[k].(*types.Document)
+		// { field: 1 } to value->field
+		if !isDoc {
 			sql += ", "
+			sql += p.Next() + "::text, _jsonb->" + p.Next()
+			args = append(args, k, k)
+			s.l.Sugar().Debugf("%s->%s", k, k)
+			continue
+		}
+
+		// field: { field: value } is not supported
+		// field: { $elemMatch: { }}
+		supportedKeys := []string{"$elemMatch"}
+		for _, fieldKey := range doc.Keys() {
+			if !slices.Contains(supportedKeys, fieldKey) {
+				s.l.Sugar().Warnf("%s not supported", fieldKey)
```

Comment:
We should return error in that case, not ignore it

Reply:
done